### PR TITLE
Change the meaning of scrub_whitelist

### PR DESCRIPTION
**Pull request by rokob**
scrub_whitelist should be a list of keys that we do not want to scrub even if they are in scrub_fields

### CODE REVIEW

**Comment by brianr** [comment]
What's the backwards-compatibility situation for this change? i.e. how will this impact customers who already have this setting in their config but then upgrade to the newest version?

**Comment by rokob** [comment]
It was introduced in 2.16.4 which was two releases ago. We don't want that behavior so I think making a breaking change over this short of a time period is a better choice than supporting the version that took a boolean. We can probably just bump the minor version, we don't have to go to 3.0 as long as it is clear what is going on in the release notes. I wouldn't expect a lot of people to be using this feature yet.

**Comment by brianr** [comment]
Sounds good 👍

File: spec/rollbar/scrubbers/params_spec.rb [hunk 1]
```diff
@@ -327,8 +327,8 @@
         let(:result) do
           [
             {
-              :foo => /\*+/,
-              :secret => 'the-secret',
+              :foo => 'bar',
+              :secret => /\*+/,
               :password => 'the-password',
               :password_confirmation => 'the-password'
```
Comment (tjallingvanderwal) [inline spec/rollbar/scrubbers/params_spec.rb]: I would expect a whitelist to only allow exact matches. Not substring/fuzzy matches.
So `scrub_whitelist = [:password]` should result in `:password_confirmation` being scrubbed.

Comment (rokob) [reply spec/rollbar/scrubbers/params_spec.rb]: Yeah that makes sense, it is a byproduct of the current way scrub_fields end up doing substring matches. The fact is if you had `password_confirmation` in the scrub_fields and put `password` in the whitelist then `password_confirmation` would get scrubbed. If you only have `password` in the scrub_fields, then both `password` and `password_confirmation` end up getting scrubbed due to the substring match, but then neither get scrubbed if you whitelist `password`.

**Comment by ArturMoczulski** [comment]
LGTM. I'm merging into `master`.

**Pull request merged**